### PR TITLE
Add tests for plugin preferences

### PR DESCRIPTION
**Pull request by bennetelli**

### CODE REVIEW

**Comment by bennetelli** [comment]
finally.. took some time to write these tests and find the tests which change the state of the preferences page components. 

We should take care and use setUp and tearDown methods when using ServiceManager.getService().set.. to change state of components because it's cached internally

**Comment by bennetelli** [comment]
What do I have to do to solve it? force push? :/

**Comment by fharms** [comment]
Do you mean the check style error is already fixed?

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java [hunk 1]
```diff
@@ -84,11 +89,42 @@ protected PsiElement getElementAtCaret() {
...
+            "Logger.",
+            "logger",
+            "appender.",
+            "rootLogger.",
+            "camel.springboot.",
+            "camel.component.",
+            "camel.dataformat.",
+            "camel.language."
+        };
+        List<String> expectedIgnoredProperties = Arrays.asList(strings1);
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java]: why not just add them directly to Arrays.asList?

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java [hunk 2]
```diff
@@ -84,11 +89,42 @@ protected PsiElement getElementAtCaret() {
...
         this.ignoreCamelCoreLib = ignoreCamelCoreLib;
     }
+
+    protected void initCamelPreferencesService() {
+        String[] strings =  {
+            "**/log4j.properties",
+            "**/log4j2.properties",
+            "**/logging.properties",
+        };
+        List<String> expectedExcludedProperties = Arrays.asList(strings);
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java]: why not just add them directly to Arrays.asList?

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CamelChosenIconCellRendererTest extends CamelLightCodeInsightFixtureTestCaseIT {
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java]: Short description of what it's suppose to test is missing

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+            "Logger.",
+            "logger",
+            "appender.",
+            "rootLogger.",
+            "camel.springboot.",
+            "camel.component.",
+            "camel.dataformat.",
+            "camel.language."
+        };
+        List<String> expectedIgnoredProperties = Arrays.asList(strings);
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java]: why not just add them directly to Arrays.asList?

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    public void testShouldContainExcludePropertyTable() {
+        CamelExcludePropertyFileTable table = camelPreferencePage.getExcludePropertyFileTable();
+        List<String> excludePropertyFiles = table.getExcludePropertyFiles();
+
+        String[] strings =  {
+            "**/log4j.properties",
+            "**/log4j2.properties",
+            "**/logging.properties",
+        };
+        List<String> expectedExcludedProperties = Arrays.asList(strings);
```
Comment (fharms) [inline camel-idea-plugin/src/test/java/org/apache/camel/idea/preference/CamelChosenIconCellRendererTest.java]: why not just add them directly to Arrays.asList?

File: camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java [hunk 6]
```diff
@@ -65,14 +69,15 @@ protected String getTestDataPath() {
      * <p>
      *   The method take a String arrays off "G:A:P:C:?" "org.apache.camel:camel-core:2.19.0"
      * </p>
-     * @param mavenAritfiact - Array of maven artifact to resolve
+     * @param mavneAritfiact - Array of maven artifact to resolve
      * @return Array of artifact files
      * @throws IOException
      */
-    private static File[] getMavenArtifacts(String... mavenAritfiact) throws IOException {
+    private static File[] getMavenArtifacts(String... mavneAritfiact) throws IOException {
```
[the remark this answers is not in the document]
Comment (bennetelli) [reply camel-idea-plugin/src/test/java/org/apache/camel/idea/CamelLightCodeInsightFixtureTestCaseIT.java]: Argh. Fixed this in a previous commit on another branch. Maybe I forgot to merge master into my branch before creating the pull request

**Comment by bennetelli** [comment]
AAAH. Got it! I checked the "organize imports" options in IntelliJ.. that's why every commit fails again and again. Maybe too much Gin Tonic for me :D

**Comment by davsclaus** [comment]
Thanks as there are 49 commits we should merge and squash them

**Comment by davsclaus** [comment]
@bennetelli maybe you can do that, eg on that green button, there is a little triangle, where you can then choose 'Squash and merge'

**Comment by bennetelli** [comment]
@davsclaus done

**Pull request merged**